[Jeremiah33-3] iP

Jeremiah33-3 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [] Managing deadlines (coming soon)
- [] Reminders (coming soon)

> [!TIP]
> If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jeremiah33-3 committed:
Add greetUser() to HeyJudy

Jeremiah33-3 committed:
Add echoCommand() to HeyJudy

Jeremiah33-3 committed:
Add the functionality to add to-do task to a list and then list it.

Jeremiah33-3 committed:
New class: add Task.java to represent class
HeyJudy.java: update static methods and add markTask to facilitate marking and unmarking of class

Task is not being represented as a class
Task should have its own class to utilise OOP principles.
Create a new class for Task so that task can store its own attributes and information, to allow data and data opeation.

Jeremiah33-3 committed:
Add derived classes of Task

Represent the task as a single Task class.
Cannot support for different types of tasks that have slightly different behaviours
Declare subclasses to Task so differentiate.

Jeremiah33-3 committed:
Add error handling logic and fix bugs in ToDo.java

Jeremiah33-3 committed:
Add the ability to delete task for users

Users cannot delete task yet
They should be able to as a user story.
Change collection of containing list of tasks to ArrayList for ease of delete and insert.

Jeremiah33-3 committed:
Add automated text ui testing function

Jeremiah33-3 committed:
Add FileManager to save task list.

Jeremiah33-3 committed:
Merge branch 'branch-Level-7'

Jeremiah33-3 committed:
Add DateTime feature and find tasks on date

Jeremiah33-3 committed:
Merge branch 'branch-Level-8'

Jeremiah33-3 committed:
Use more OOP in the bot design

Jeremiah33-3 committed:
Merge branch 'branch-A-MoreOOP'

Jeremiah33-3 committed:
Package the classes

Jeremiah33-3 committed:
Remove an unused import in HeyJudy

Jeremiah33-3 committed:
Merge branch 'branch-A-Packages'

Jeremiah33-3 committed:
Merge branch 'add-gradle-support'

Jeremiah33-3 committed:
Add Gradle for automation

Jeremiah33-3 committed:
Add JUnit tests

Jeremiah33-3 committed:
Package the App as a JAR file

Jeremiah33-3 committed:
Tweak the code to comply with a coding standard

Jeremiah33-3 committed:
Update findCommand to search by words

Jeremiah33-3 committed:
Add JavaDoc comments

Jeremiah33-3 committed:
Merge branch 'branch-A-CodingStandard'

Jeremiah33-3 committed:
Merge branch 'branch-Level-9'

Jeremiah33-3 committed:
Resolve merge conflicts in Task, Event, Deadline

Jeremiah33-3 committed:
Use CheckStyle

Jeremiah33-3 committed:
Merge branch 'branch-A-CheckStyle'

Jeremiah33-3 committed:
Adhere code to coding standard via CheckStyle

Jeremiah33-3 committed:
Merge branch 'branch-A-CheckStyle'

Jeremiah33-3 committed:
Add GUI with JavaFX

Jeremiah33-3 committed:
Merge branch 'branch-Level-10'

Jeremiah33-3 committed:
Fix logic issues in isDateFormat method in FindCommand

Jeremiah33-3 committed:
Add varrags keywords search in FindCommand

Jeremiah33-3 committed:
Merge branch 'branch-A-Varargs'

NHT020305 commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,151 @@
+    public String getTo() {
+        return this.to;
+    }
+}
```
I like these header comments!

NHT020305 commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,126 @@
+    public String getQuery() {
+        return this.query;
+    }
+}
```
should we put header comments for those private methods since they are non-trivial private method ?

NHT020305 commented on src/test/java/storage/StorageTest.java:
```diff
@@ -0,0 +1,158 @@
+        Task corruptedTask = storage.parseTaskFromFile("Invalid | Format | Here");
+        assertNull(corruptedTask, "Parsing a corrupted line should return null");
+    }
+}
```
Nice comment to make the code more understandable!

NHT020305 reviewed (commented):
Overall, your code looks very good to me

NHT020305 commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,126 @@
+
+        return getMatchingTasksString(matchingTasks, "Here are the matching tasks in your list:");
+    }
+
```
should we put header comments for those private methods since I think they are non-trivial private methods?

NHT020305 commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,126 @@
+            return searchByKeyword(tasks, keywords);
+        }
+    }
+
```
Should we put head comments for those private methods because I think they are non-trivial private method ?

NHT020305 reviewed (commented):
Overall, the code looks very good to me.

Jeremiah33-3 committed:
Use Java Assertions

There is no assert statements to document assumptions.

Assertions help to define assumptions in programs so that runtime can verify them.

Added assert statements to point of code where certain program states are being assumed.

Jeremiah33-3 committed:
Improve Code Quality

There are some codes that do not adhere to code quality.

Codes should adhere to code quality to improve readability.

Refactored some long methods and codes with weak SLAP.

Jeremiah33-3 committed:
Use Streams to enhance code

Jeremiah33-3 committed:
Merge pull request #4 from Jeremiah33-3/branch-A-Assertions

Use Java Assertions

Jeremiah33-3 committed:
Merge pull request #5 from Jeremiah33-3/branch-A-CodeQuality

Refactor code to improve code quality

Jeremiah33-3 committed:
Merge pull request #6 from Jeremiah33-3/branch-A-Streams

Use Java Streams

Jeremiah33-3 committed:
Add function to manage recurring task

We can only manage nonrecurring tasks with the app.

Users might want to view their recurring tasks neatly and conveniently too.

Add a new class RecurringTask to represent the abstraction of tasks repeating by a frequency.

Currently, can track frequency of daily, weekly, and monthly. In the future, possible to extend.

Jeremiah33-3 committed:
Merge branch 'branch-B-RecurringTasks'

Jeremiah33-3 committed:
Add more error handling logic to the classes

Jeremiah33-3 committed:
Merge branch 'branch-A-MoreErrorHandling'

Jeremiah33-3 committed:
Add more automated tests

Jeremiah33-3 committed:
Merge branch 'branch-A-MoreTesting'

Jeremiah33-3 committed:
Improve GUI

Jeremiah33-3 committed:
Merge branch 'branch-A-BetterGui'

Jeremiah33-3 committed:
Update README.md

Jeremiah33-3 committed:
Update Ui.png

Jeremiah33-3 committed:
Merge branch 'master' of https://github.com/Jeremiah33-3/ip

Jeremiah33-3 committed:
Write User Guide

Jeremiah33-3 committed:
Add more error handling

Jeremiah33-3 committed:
Update README.md

Jeremiah33-3 committed:
Update README.md

Jeremiah33-3 committed:
Fix date parsing issue

Jeremiah33-3 committed:
Merge branch 'master' of https://github.com/Jeremiah33-3/ip
